src/core/bittorrent/session.cpp

```diff
@@ -2111,6 +2122,9 @@ void Session::handleAddTorrentAlert(libtorrent::add_torrent_alert *p)
             }
         }
 
+        if (pref->isAddTrackersEnabled() && !torrent->isPrivate())
+            torrent->addTrackers(m_additionalTrackers);
```

Comment:
I don't think this works when we assign a torrent_info object...
I haven't tested it, this is what the docs seem to say implicitly.